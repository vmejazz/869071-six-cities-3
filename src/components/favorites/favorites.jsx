import React, {PureComponent} from "react";
import UserProfile from "../user-profile/user-profile.jsx";
import {Link} from "react-router-dom";
import {connect} from "react-redux";
import {Operation} from "../../reducer/data/data.js";
import {getOffersFavorite, getCityesFavorite} from "../selectors.js";
import FavoritesList from "../favorites-list/favorites-list.jsx";
import PropTypes from "prop-types";
import FavoritesEmpty from "../favorites-empty/favorites-empty.jsx";
class Favorites extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {loadFavorites} = this.props;
    loadFavorites();
  }

  componentDidUpdate() {
    // const {loadFavorites} = this.props;
    // loadFavorites();
  }

  // loadFavorites();
  render() {
    const {offersFavorite, cityesFavorite} = this.props;

    let isEmptyList = offersFavorite.length < 1;

    return (
      <div className="page">
        <header className="header">
          <div className="container">
            <div className="header__wrapper">
              <div className="header__left">
                <Link className="header__logo-link" to="/">
                  <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width={81} height={41} />
                </Link>
              </div>
              <nav className="header__nav">
                <ul className="header__nav-list">
                  <li className="header__nav-item user">
                    <UserProfile />
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </header>
        <main className="page__main page__main--favorites">
          <div className="page__favorites-container container">
            {isEmptyList ?
              <FavoritesEmpty />
              : <section className="favorites">
                <h1 className="favorites__title">Saved listing</h1>
                <FavoritesList
                  offersFavorite={offersFavorite}
                  cityesFavorite={Object.keys(cityesFavorite)}
                />
              </section>
            }
          </div>
        </main>
        <footer className="footer container">
          <a className="footer__logo-link" href="main.html">
            <img className="footer__logo" src="img/logo.svg" alt="6 cities logo" width={64} height={33} />
          </a>
        </footer>
      </div>
    );
  }
}

Favorites.propTypes = {
  offersFavorite: PropTypes.array,
  cityesFavorite: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object
  ]),
  loadFavorites: PropTypes.func
};

const mapStateToProps = (state) => ({
  // activeCity: state.DATA.activeCity,
  // hoverCardId: state.DATA.hoverCardId,
  // userInfo: getUser(state),
  // cityes: getCityes(state),
  // offersShow: getOffersShow(state),
  offersFavorite: getOffersFavorite(state),
  cityesFavorite: getCityesFavorite(state),
});

const mapDispatchToProps = (dispatch) => ({
  loadFavorites() {
    dispatch(
        Operation.loadFavorites()
    );
  },
});

export {Favorites};
export default connect(mapStateToProps, mapDispatchToProps)(Favorites);

