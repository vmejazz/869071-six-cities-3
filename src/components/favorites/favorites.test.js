import React from "react";
import renderer from "react-test-renderer";
import {Favorites} from "./favorites.jsx";
import customHistory from "../../history.js";
import {Router} from "react-router-dom";
jest.mock(`../bookmark-button/bookmark-button.jsx`, () => `BookmarkButton`);
jest.mock(`../user-profile/user-profile.jsx`, () => `UserProfile`);
jest.mock(`../reviews-list/reviews-list.jsx`, () => `ReviewsList`);
jest.mock(`../map/map.jsx`, () => `Map`);
jest.mock(`../apartment-list/apartment-list.jsx`, () => `ApartmentList`);
jest.mock(`../reviews-form/reviews-form.jsx`, () => `ReviewsForm`);

const offersFavorite = [{
  id: 1,
  title: `Beautiful apartment`,
  price: 120,
  srcImg: `img/apartment-01.jpg`,
  imageURLs: [`img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`],
  description: `All super long text about apartmmnet with cools stuffs`,
  premium: true,
  type: `Room`,
  rate: 4.8,
  bedrooms: 3,
  maxGuests: 4,
  apartmentDetails: [`Wifi`, `Cable TV`, `Kitchen`],
  ownerInfo: {
    name: `Jon`,
    super: true,
    srcAvatar: `img/avatar-max.jpg`
  },
  positions: [234, 234234]
}];

const cityesFavorite = {
  Dusseldorf: {

  },
  Amsterdam: {

  },
  Cologne: {

  }
};
const loadFavorites = () => {};

it(`<Favorites /> should render favorites page `, () => {
  const tree = renderer
      .create(
          <Router history={customHistory} >
            <Favorites
              offersFavorite={offersFavorite}
              cityesFavorite={cityesFavorite}
              loadFavorites={loadFavorites}
            />
          </Router>
      )
      .toJSON();

  expect(tree).toMatchSnapshot();
});
