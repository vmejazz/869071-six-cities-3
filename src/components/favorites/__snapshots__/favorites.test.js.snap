// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Favorites /> should render favorites page  1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height={41}
              src="img/logo.svg"
              width={81}
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <UserProfile />
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="/"
                  onClick={[Function]}
                >
                  <span>
                    Dusseldorf
                  </span>
                </a>
              </div>
            </div>
            <ApartmentList
              isFavoriteList={true}
              offersShow={Array []}
              onCardHover={[Function]}
            />
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="/"
                  onClick={[Function]}
                >
                  <span>
                    Amsterdam
                  </span>
                </a>
              </div>
            </div>
            <ApartmentList
              isFavoriteList={true}
              offersShow={Array []}
              onCardHover={[Function]}
            />
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="/"
                  onClick={[Function]}
                >
                  <span>
                    Cologne
                  </span>
                </a>
              </div>
            </div>
            <ApartmentList
              isFavoriteList={true}
              offersShow={Array []}
              onCardHover={[Function]}
            />
          </li>
        </ul>
      </section>
    </div>
  </main>
  <footer
    className="footer container"
  >
    <a
      className="footer__logo-link"
      href="main.html"
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height={33}
        src="img/logo.svg"
        width={64}
      />
    </a>
  </footer>
</div>
`;
